clinica/pydra/pet_volume/pipeline.py

```diff
@@ -0,0 +1,306 @@
+from nipype.interfaces.petpvc import PETPVC
+from nipype.interfaces.spm import Coregister, DARTELNorm2MNI, Smooth
+from nipype.interfaces.spm.utils import Reslice
+from nipype.algorithms.misc import Gunzip
+import pydra
+from pydra.engine import Workflow
+from pydra.tasks.nipype1.utils import Nipype1Task
+from typing import Any
+from clinica.pydra.engine import clinica_io
+from clinica.pydra.pet_volume.tasks import (
+    apply_binary_mask,
+    atlas_statistics,
+    create_binary_mask,
+    create_pvc_mask,
+    normalize_to_reference,
+)
+
+
+def _check_pipeline_parameters(parameters: dict) -> dict:
+    """Check the parameters passed to the pipeline.
+
+    Parameters
+    ----------
+    parameters : dict
+        Dictionary of parameters to analyze.
+
+    Returns
+    -------
+    dict :
+        Cleaned dictionary of parameters.
+    """
+    from clinica.utils.atlas import PET_VOLUME_ATLASES
+    from clinica.utils.group import check_group_label
+
+    parameters.setdefault("group_label", None)
+    check_group_label(parameters["group_label"])
+    if "acq_label" not in parameters.keys():
+        raise KeyError("Missing compulsory acq_label key in pipeline parameter.")
+    parameters.setdefault("pvc_psf_tsv", None)
+    parameters["apply_pvc"] = parameters["pvc_psf_tsv"] is not None
+    parameters.setdefault("mask_tissues", [1, 2, 3])
+    parameters.setdefault("mask_threshold", 0.3)
+    parameters.setdefault("pvc_mask_tissues", [1, 2, 3])
+    parameters.setdefault("smooth", [8])
+    parameters.setdefault("atlases", PET_VOLUME_ATLASES)
+    return parameters
+
+
+@clinica_io
+def build_core_workflow(name: str = "core", parameters: dict = {}) -> Workflow:
+    """Core workflow for the PET Volume pipeline.
+
+    Parameters
+    ----------
+    name : The name of the workflow.
+
+    Returns
+    -------
+    workflow : The core workflow.
+    """
+    from clinica.utils.spm import spm_standalone_is_available, use_spm_standalone
+
+    if spm_standalone_is_available():
+        use_spm_standalone()
+
+    parameters = _check_pipeline_parameters(parameters)
+
+    input_spec = pydra.specs.SpecInfo(
+        name="Input",
+        fields=[
+            ("_graph_checksums", Any),
+            ("T1w", str, {"mandatory": True}),
+            ("pet", str, {"mandatory": True}),
+            (
+                "mask_tissues",
+                dict,
+                {"tissue_number": 1, "modulation": False},
+                {"mandatory": True},
+            ),
+            ("flow_fields", dict, {"group_label": "group"}, {"mandatory": True}),
```

Comment:
The name of the `group_label` should not be hardcoded but instead be a user input which depends on the name provided when running `t1_volume`

Reply:
Yes, I fixed that already locally but didn't commit it yet.